langchain_tools_demo/main.py

```diff
@@ -49,19 +36,17 @@ async def lifespan(app: FastAPI):
     print("Loading application...")
     yield
     # FastAPI app shutdown event
-    close_client_tasks = [
-        asyncio.create_task(a.client.close()) for a in user_agents.values()
-    ]
-
-    asyncio.gather(*close_client_tasks)
+    app.state.orchestrator.close_clients()
 
 
 @routes.get("/")
 @routes.post("/")
 async def index(request: Request):
     """Render the default template."""
     # Agent setup
-    agent = await get_agent(request.session, user_id_token=None)
+    orchestrator = request.app.state.orchestrator
+    await orchestrator.user_session_create(request.session)
+    templates = Jinja2Templates(directory="templates")
```

Comment:
nit: this should be global state since it's static. We don't want to reload on every call to index.